umbral/_pre.py

```diff
@@ -112,19 +114,22 @@ def verify_kfrag(kfrag,
 
     u = params.u
 
+    id = kfrag._id
+    key = kfrag._bn_key
     u1 = kfrag._point_commitment
     z1 = kfrag._bn_sig1
     z2 = kfrag._bn_sig2
-    x = kfrag._point_noninteractive
-    key = kfrag._bn_key
+    ni = kfrag._point_noninteractive
+    xcoord = kfrag._point_xcoord
+    
 
     #  We check that the commitment u1 is well-formed
     correct_commitment = u1 == key * u
 
     # We check the Schnorr signature over the kfrag components
     g_y = (z2 * params.g) + (z1 * pubkey_a_point)
 
-    kfrag_components = [g_y, kfrag._bn_id, pubkey_a_point, pubkey_b_point, u1, x]
-    valid_kfrag_signature = z1 == CurveBN.hash(*kfrag_components, params=params)
+    signature_input = [g_y, id, pubkey_a_point, pubkey_b_point, u1, ni, xcoord]
+    valid_kfrag_signature = z1 == CurveBN.hash(*signature_input, params=params)
```

Comment:
You can actually just do:

`valid_kfrag_signature = z1 == CurveBN.hash(g_y, id, pubkey_a_point, pubkey_b_point, u1, ni, xcoord, params=params)`

This avoids making the list first.

Reply:
I know, but the list is becoming a little bit long, plus I wanted to stress what's what we are signing.

Reply:
Yeah, well I also wondered why we were making it a list here - it never mutates.  Why not a tuple?